src/devices/Amg88xx/Amg88xx.cs

```diff
@@ -0,0 +1,475 @@
+
+            Span<byte> buffer = stackalloc byte[Rows * Columns * BytesPerPixel];
+            _i2cDevice.Read(buffer);
+
+            var image = new int[Columns, Rows];
+
+            int idx = 0;
+            for (int r = 0; r < Rows; r++)
+            {
+                for (int c = 0; c < Columns; c++)
+                {
+                    byte tl = buffer[idx++];
+                    byte th = buffer[idx++];
+                    image[c, r] = th << 8 | tl;
+                }
+            }
+
+            return image;
+        }
+
+        /// <summary>
+        /// Gets the temperature reading from the internal thermistor.
+        /// </summary>
+        /// <returns>Temperature reading</returns>
+        public Temperature GetSensorTemperature()
+        {
+            byte tthl = GetRegister(Register.TTHL);
+            byte tthh = GetRegister(Register.TTHH);
+
+            int reading = (tthh & 0x7) << 8 | tthl;
+            reading = tthh >> 3 == 0 ? reading : -reading;
+
+            // The LSB is equivalent to 0.0625℃.
+            return Temperature.FromDegreesCelsius(reading * ThermistorTemperatureResolution);
+        }
+
+        #endregion
+
+        #region Status
+
+        /// <summary>
+        /// Gets the temperature overflow flag from the status register
+        /// </summary>
+        /// <returns>Temperature overflow flag</returns>
+        public bool HasTemperatureOverflow()
+        {
+            return GetBit(Register.STAT, (byte)StatusFlagBit.OVF_IRS);
+        }
+
+        /// <summary>
+        /// Clears the temperature overflow flag in the status register
+        /// </summary>
+        public void ClearTemperatureOverflow()
+        {
+            // only the bit to be cleared is set, the other bits need to be 0
+            SetRegister(Register.SCLR, 1 << (byte)StatusClearBit.OVFCLR);
+        }
+
+        /// <summary>
+        /// Gets the thermistor overflow flag from the status register.
+        /// Note: the bit is only menthioned in early versions of the reference specification.
+        /// It is not clear whether this is a specification error or a change in a newer
+        /// revision of the sensor.
+        /// </summary>
+        /// <returns>Thermistor overflow flag</returns>
+        public bool HasThermistorOverflow()
+        {
+            return GetBit(Register.STAT, (byte)StatusFlagBit.OVF_THS);
+        }
+
+        /// <summary>
+        /// Clears the thermistor overflow flag in the status register
+        /// </summary>
+        public void ClearThermistorOverflow()
+        {
+            // only the bit to be cleared is set, the other bits need to be 0
+            SetRegister(Register.SCLR, 1 << (byte)StatusClearBit.OVFTHCLR);
+        }
+
+        /// <summary>
+        /// Gets the interrupt flag from the status register
+        /// </summary>
+        /// <returns>Interrupt flag</returns>
+        public bool HasInterrupt()
```

Comment:
when and how should users use this? Should you have an API to make this usage better instead of making this public directly? I.e. WaitForXYZAsync with while loop waiting for interrupt and and clearing it?

Reply:
@krwq I think you suggest a slightly higher level of abstraction from the sensor details. 
I still see a value in making the "pure" interrupt information available to the user as this allows to check at any time whether the sensor has seen a temperature which exceeded the configured lower or uppter threshold of any pixel. 
Though, I'll think about adding some "semantics" to the API naming. 
I'll also think about adding a WaitForSomethingAsync in addition.

Reply:
fair enough for now, if we have users asking and clearer idea of API shape we can improve in the future.

Reply:
@krwq Even though I can't promisse anything, I'll keep an eye on that and may pick up future improvement requests.